### ISSUE TITLE
PyroModule.requires_grad_(False).__call__() issues unwanted warning

### ISSUE THREAD

**Issue by fritzo** [pyro-ppl/pyro#2705]
After training a `PyroModule` model, I'd like to detach it and use it elsewhere in a "frozen" state. One obvious way to do so would be to call `my_module.requires_grad_(False)` to detach all parameters. Doing so results in the desired behavior, but has the side effect of triggering the following warning (many many times):
```
/Users/fobermey/github/pyro-ppl/pyro/pyro/primitives.py:361: UserWarning: linear.bias was not registered in the param store because requires_grad=False
  " requires_grad=False")
```

It is unclear to me how to resolve this. Should we provide a `detach_()` method for modules that converts all `PyroParam`s to `Tensor`s? Should we simply remove that warning if it occurs in a `PyroModule`? Should we be calling some other method like `my_module.train(False)`?
